File: CS/MyRadioGroup/ExtendedEditor/MyRadioGroup.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraEditors;
using System.ComponentModel;

namespace EditorDescendant
{
    [System.ComponentModel.DesignerCategory("")]
    public class MyRadioGroup : RadioGroup
    {

        #region Service
        static MyRadioGroup()
        {
            MyRepositoryItemRadioGroup.Register();
        }

        public MyRadioGroup()
        { }

        public override string EditorTypeName
        { get { return MyRepositoryItemRadioGroup.EditorName; } }
        
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new MyRepositoryItemRadioGroup Properties
        { get { return base.Properties as MyRepositoryItemRadioGroup; } }


     #endregion
    }
}

File: VB/MyRadioGroup/ExtendedEditor/MyRadioGroup.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraEditors
Imports System.ComponentModel

Namespace EditorDescendant
	<System.ComponentModel.DesignerCategory("")> _
	Public Class MyRadioGroup
		Inherits RadioGroup

		#Region "Service"
		Shared Sub New()
			MyRepositoryItemRadioGroup.Register()
		End Sub

		Public Sub New()
		End Sub

		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return MyRepositoryItemRadioGroup.EditorName
			End Get
		End Property

		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Shadows ReadOnly Property Properties() As MyRepositoryItemRadioGroup
			Get
				Return TryCast(MyBase.Properties, MyRepositoryItemRadioGroup)
			End Get
		End Property


	 #End Region
	End Class
End Namespace
